bitcoin/chainparams.c

```diff
@@ -123,3 +123,12 @@ const struct chainparams *chainparams_by_bip173(const char *bip173_name)
 	}
 	return NULL;
 }
+
+const char *chainparams_get_network_names(const tal_t *ctx)
+{
+    char* networks_string = tal_fmt(ctx, "%s", networks[0].network_name);
```

Comment:
Sorry for the flip-flop, but I realized just now that this could be simpler done as `tal_strdup(ctx, networks[0].network_name)`.

`tal_append_fmt` for the loop is fine.

Also, nitpick, it would be nice to `STATIC_ASSERT(ARRAY_SIZE(networks) != 0);` before this, but then we would not be supporting any networks in that case and be useless, so maybe pointless to do that, you can probably ignore this bit.

Nitpick:  we use `char *networks_string` not `char* networks_string` in this project (I agree with the latter personally, but this project is pretty consistent with binding asterisks to the variable name).